atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifierTest.java

```diff
@@ -0,0 +1,264 @@
+    private static final String HOST_2 = "host2";
+    private static final String HOST_3 = "host3";
+    private static final String HOST_4 = "host4";
+    private static final String RACK_1 = "test_rack1";
+    private static final String RACK_2 = "test_rack2";
+
+    private CassandraClient client = mock(CassandraClient.class);
+    private CassandraKeyValueServiceConfig config = mock(CassandraKeyValueServiceConfig.class);
+
+    @Test
+    public void unsetTopologyAndHighRfThrows() throws TException {
+        setTopology(defaultTopology(HOST_1), defaultTopology(HOST_2));
+        when(config.replicationFactor()).thenReturn(3);
+
+        assertThatThrownBy(() -> CassandraVerifier.sanityCheckDatacenters(client, config))
+                .isInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    public void unsetTopologyAndRfOneSucceeds() throws TException {
+        setTopology(defaultTopology(HOST_1), defaultTopology(HOST_2));
+        when(config.replicationFactor()).thenReturn(1);
+
+        assertThat(CassandraVerifier.sanityCheckDatacenters(client, config))
+                .containsExactly(CassandraConstants.DEFAULT_DC);
+    }
+
+    @Test
+    public void nonDefaultDcAndHighRfSucceeds() throws TException {
+        setTopology(createDetails(DC_1, RACK_1, HOST_1));
+        when(config.replicationFactor()).thenReturn(3);
+
+        assertThat(CassandraVerifier.sanityCheckDatacenters(client, config))
+                .containsExactly(DC_1);
+    }
+
+    @Test
+    public void oneDcOneRackAndMoreHostsThanRfThrows() throws TException {
+        setTopology(createDetails(DC_1, RACK_1, HOST_1),
+                createDetails(DC_1, RACK_1, HOST_2),
+                createDetails(DC_1, RACK_1, HOST_3),
+                createDetails(DC_1, RACK_1, HOST_4));
+        when(config.replicationFactor()).thenReturn(3);
+
+        assertThatThrownBy(() -> CassandraVerifier.sanityCheckDatacenters(client, config))
+                .isInstanceOf(IllegalStateException.class);
+    }
+
+    // This test was failing prior to refactor
+    @Test
+    public void oneDcFewerRacksThanRfAndMoreHostsThanRfThrows() throws TException {
+        setTopology(defaultDcDetails(RACK_1, HOST_1),
+                defaultDcDetails(RACK_2, HOST_2),
+                defaultDcDetails(RACK_1, HOST_3),
+                defaultDcDetails(RACK_2, HOST_4));
+        when(config.replicationFactor()).thenReturn(3);
+
+        assertThatThrownBy(() -> CassandraVerifier.sanityCheckDatacenters(client, config))
+                .isInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    public void oneDcMoreRacksThanRfAndMoreHostsThanRfSucceeds() throws TException {
+        setTopology(defaultDcDetails(RACK_1, HOST_1),
+                defaultDcDetails(RACK_2, HOST_2),
+                defaultDcDetails(RACK_1, HOST_3),
+                defaultDcDetails(RACK_2, HOST_4));
+        when(config.replicationFactor()).thenReturn(2);
```

Comment:
nit: do we want a third rack? 2 racks isn't more than an RF of 2

Reply:
University just ruined me to use greater as greater or equal :P